File: 2026/07-July/12-Sunday/solution.js
var arrayRankTransform = function(arr) {
    const sorted = [...new Set(arr)].sort((a, b) => a - b);

    const rank = new Map();

    sorted.forEach((num, idx) => {
        rank.set(num, idx + 1);
    });

    return arr.map(num => rank.get(num));
};

File: 2026/07-July/12-Sunday/solution.cs
public class Solution {
    public int[] ArrayRankTransform(int[] arr) {
        int[] sorted = (int[])arr.Clone();
        Array.Sort(sorted);

        Dictionary<int, int> rank = new Dictionary<int, int>();

        foreach (int num in sorted) {
            if (!rank.ContainsKey(num))
                rank[num] = rank.Count + 1;
        }

        int[] ans = new int[arr.Length];

        for (int i = 0; i < arr.Length; i++) {
            ans[i] = rank[arr[i]];
        }

        return ans;
    }
}
